core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -219,23 +218,6 @@ class GameInfo {
         // which in turn leads to us trying to get info on all the building in all the cities...
         // which can fail if there's an "unregistered" building anywhere
         for (civInfo in civilizations) {
-            for (cityInfo in civInfo.cities) {
-                val cityConstructions = cityInfo.cityConstructions
-
-                // As of 2.9.6, removed hydro plant, since it requires rivers, which we do not yet have
-                if ("Hydro Plant" in cityConstructions.builtBuildings)
-                    cityConstructions.builtBuildings.remove("Hydro Plant")
```

Comment:
That's not what this code is for.
This code is for validating saved games, if they're not in sync with the game files the game will crash.
It's possible that the time has come for us to remove some of the validations, because there are no longer such save files, but that's an entirely separate discussion.